src/components/TokenEnterAmount.tsx

```diff
@@ -275,7 +273,16 @@ export function useEnterAmount(props: {
       return
     }
 
-    const rawValue = unformatNumberForProcessing(value)
+    /**
+     * If the value comes from external source (e.g. API response) then it will probably be a
+     * regular JS-compatible number in a format "1234.5678" for which we should skip the
+     * unformatting step.
+     */
```

Comment:
fwiw i don't think this comment is necessary since we provide context in the `unformatNumberForProcessing` function. i also worry a little that this comment is overly specific - if i understand correctly, we want to avoid unformatting numbers that are already formatted in a standard way, otherwise we risk unformatting using the device format settings. this could be true regardless of whether the value comes from an API response?